src/planner/planner.c

```diff
@@ -383,6 +383,71 @@ preprocess_query(Node *node, PreprocessQueryContext *context)
 		Index rti = 1;
 		bool ret;
 
+		/*
+		 * Detect FOREIGN KEY lookup queries and mark the RTE for expansion.
+		 * Unfortunately postgres will create lookup queries for foreign keys
+		 * with `ONLY` preventing hypertable expansion. Only for declarative
+		 * partitioned tables the queries will be created without `ONLY`.
+		 * We try to detect these queries here and undo the `ONLY` flag for
+		 * these specific queries.
+		 *
+		 * The implementation of this on the postgres side can be found in
+		 * src/backend/utils/adt/ri_triggers.c
+		 */
+
+		/*
+		 * RI_FKey_check
+		 *
+		 * The RI_FKey_check query string built is
+		 *  SELECT 1 FROM [ONLY] <pktable> x WHERE pkatt1 = $1 [AND ...]
+		 *       FOR KEY SHARE OF x
+		 */
+		if (query->commandType == CMD_SELECT && query->hasForUpdate &&
+			list_length(query->rtable) == 1 && context->root->glob->boundParams)
```

Comment:
Isn't it break if someone try a similar query for another purpose??? I mean, for hypertables make no sense a `FROM ONLY` query... but who knows!!!